frontend/javascripts/oxalis/model/sagas/quick_select_ml_saga.ts

```diff
@@ -0,0 +1,279 @@
+    point_labels: new ort.Tensor("float32", onnxLabel, [1, 2]),
+    mask_input: new ort.Tensor("float32", onnxMaskInput, [1, 1, 256, 256]),
+    has_mask_input: new ort.Tensor("float32", onnxHasMaskInput, [1]),
+    orig_im_size: new ort.Tensor("float32", origImSize, [2]),
+  };
+  const { masks, iou_predictions: iouPredictions } = await ortSession.run(ortInputs);
+
+  // @ts-ignore
+  const bestMaskIndex = iouPredictions.data.indexOf(Math.max(...iouPredictions.data));
+  const maskData = new Uint8Array(EMBEDDING_SIZE[0] * EMBEDDING_SIZE[1]);
+  const startOffset = bestMaskIndex * EMBEDDING_SIZE[0] * EMBEDDING_SIZE[1];
+  for (let idx = 0; idx < EMBEDDING_SIZE[0] * EMBEDDING_SIZE[1]; idx++) {
+    maskData[idx] = masks.data[idx + startOffset] > 0 ? 1 : 0;
+  }
+
+  const size = embeddingBoxInTargetMag.getSize();
+  const userSizeInTargetMag = userBoxInTargetMag.getSize();
+  // Somewhere between the front-end, the back-end and the embedding
+  // server, there seems to be a different linearization of 2D the image
+  // data which is why the code here deals with the XZ plane as a special
+  // case.
+  const stride =
+    activeViewport === "PLANE_XZ"
+      ? [size[1], size[0], size[0] * size[1] * size[2]]
+      : [size[2], size[0], size[0] * size[1] * size[2]];
+
+  let mask = ndarray(maskData, size, stride);
+  mask = mask
+    // a.lo(x,y) => a[x:, y:]
+    .lo(topLeft[firstDim], topLeft[secondDim], 0)
+    // a.hi(x,y) => a[:x, :y]
+    .hi(userSizeInTargetMag[firstDim], userSizeInTargetMag[secondDim], 1);
+  return mask;
+}
+
+export function* prefetchEmbedding(action: MaybePrefetchEmbeddingAction) {
+  const preparation = yield* call(prepareQuickSelect, action, true);
+  if (preparation == null) {
+    return;
+  }
+  const { labeledResolution, activeViewport, colorLayer } = preparation;
+  const { startPosition } = action;
+  const PREFETCH_WINDOW_SIZE = [100, 100, 0] as Vector3;
+  const endPosition = V3.add(
+    startPosition,
+    Dimensions.transDim(PREFETCH_WINDOW_SIZE, activeViewport),
+  );
+
+  const userBoxMag1 = new BoundingBox({
+    min: V3.floor(V3.min(startPosition, endPosition)),
+    max: V3.floor(
+      V3.add(V3.max(startPosition, endPosition), Dimensions.transDim([0, 0, 1], activeViewport)),
+    ),
+  });
+
+  console.time("prefetch session and embedding");
+  const dataset = yield* select((state: OxalisState) => state.dataset);
+  // Won't block, because the return value is not a promise (but contains
+  // a promise instead)
+  const { embeddingPromise } = yield* call(
+    getEmbedding,
+    dataset,
+    colorLayer.name,
+    userBoxMag1,
+    labeledResolution,
+    activeViewport,
+  );
+  // Also prefetch session (will block). After the first time, it's basically
+  // a noop.
+  yield* call(getSession);
+
+  try {
+    // Await the promise here so that the saga finishes once the embedding was loaded
+    // (this simplifies debugging and time measurement).
+    yield* call(() => embeddingPromise);
```

Comment:
Does this offer an advantage to
```suggestion
    yield embeddingPromise;
```
?